### DISCUSSION TITLE
Saving fixtures to file

### DISCUSSION THREAD

**Discussion by 5HT2** [uptrace/bun#124] [Q&A]
Hello, I've created queries for inserting users into my database, loosely based off of the basic example in the repository. I've been struggling to find how I'd actually save my database to a fixture file now, or just saving it overall. It doesn't seem to be in the docs, nor can I find the answer to this in the existing issues..

**Comment by 5HT2** [comment]
I came up with this workaround but it seems really weird that there wouldn't be a built-in function for this
```go
func testDB() {
	users := make([]User, 0)
	if err := db.NewSelect().Model(&users).OrderExpr("id ASC").Scan(ctx); err != nil {
		panic(err)
	}

	if *debug {
		log.Printf("Users: %v", users)
	}

	formattedData := []fixtureData{{Model: "User", Rows: users}}
	data, err := yaml.Marshal(formattedData)
	if err != nil {
		panic(err)
	}

	err = ioutil.WriteFile("config/test.yaml", data, fs.FileMode(0700))

	if err != nil {
		panic(err)
	}
}
```

**Comment by vmihailenco** [comment]
I agree that there should be some builtin function, but I am not sure about the API. Perhaps something like:

```go
enc := dbfixture.NewEncoder(writer)
if err := enc.Encode(users); err != nil {
    panic(err)
}
```

And it will not be much/any better than your function...

**Reply by 5HT2** [reply]
Well given the fact there's a `fixture.Load` method, a `fixture.Save` method would've seemed natural. 
At least a built-in method would do the yaml encoding for you, instead of the user having to make their own structs and such...

I'm actually not sure how anyone else is supposed to be saving their DB if such a method is missing

**Reply by vmihailenco** [reply]
It is a good point about `Load`, but it uses `fs.FS` which does not support creating files in an abstract way. So I went with https://github.com/uptrace/bun/pull/130

>I'm actually not sure how anyone else is supposed to be saving their DB if such a method is missing

Presumably most of the time you write fixtures by hands. It is test data you want to get into a db, not to save it from a db. But I guess sometimes it is handy to save a database into a file...
